apps/aehttp/test/aehttp_integration_SUITE.erl

```diff
@@ -0,0 +1,657 @@
+    %{ok, []} = rpc(aec_tx_pool, peek, [infinity]), % empty again
+    %{ok, 200, []} = get_transactions(), 
+
+    %{ok, 200, #{<<"balance">> := Bal1}} = get_balance(),  
+    %Bal1 = Bal0 + MineReward + Fee - AmountToSpent - Fee,
+    %{ok, 200, #{<<"balance">> := RecBalance}} = 
+    %             get_balance(base64:encode(ReceiverPubKey)),
+
+    ok.
+
+post_correct_blocks(Config) ->
+    BlocksToPost = 10,
+    aecore_suite_utils:mine_blocks(aecore_suite_utils:node_name(?NODE),
+                                   BlocksToPost),
+    Blocks =
+        lists:map(
+            fun(Height) ->
+                {ok, B} = rpc(aec_conductor, get_block_by_height, [Height]),
+                B
+            end,
+            lists:seq(1, BlocksToPost)),
+    % consider a rpc call for cleaning the aec_chain_state's state
+    aecore_suite_utils:stop_node(?NODE, Config),
+    aecore_suite_utils:start_node(?NODE, Config),
+    aecore_suite_utils:connect(aecore_suite_utils:node_name(?NODE)),
+    GH = rpc(aec_conductor, top_header, []), 
+    0 = aec_headers:height(GH), %chain is empty
+    lists:foreach(
+        fun(Block) ->
+            {ok, 200, _} = post_block(Block),
+            H = rpc(aec_conductor, top_header, []), 
+            {ok, HH} = aec_headers:hash_header(H),
+            {ok, BH} = aec_blocks:hash_internal_representation(Block),
+            BH = HH % block accepted
+        end,
+        Blocks),
+    ok.
+
+post_broken_blocks(Config) ->
+    aecore_suite_utils:mine_blocks(aecore_suite_utils:node_name(?NODE),
+                                   1),
+    {ok, CorrectBlock} = rpc(aec_conductor, get_block_by_height, [1]),
+    % consider a rpc call for cleaning the aec_chain_state's state
+    aecore_suite_utils:stop_node(?NODE, Config),
+    aecore_suite_utils:start_node(?NODE, Config),
+    aecore_suite_utils:connect(aecore_suite_utils:node_name(?NODE)),
+    GH = rpc(aec_conductor, top_header, []), 
+    0 = aec_headers:height(GH), %chain is empty
+    CorrectBlockMap = aec_blocks:serialize_to_map(CorrectBlock),
+    BrokenBlocks =
+        lists:map(
+            fun({Key, Value}) ->
+                BrokenBlock = maps:put(Key, Value, CorrectBlockMap),
+                {Key, BrokenBlock}
+            end,
+        [{<<"prev_hash">>, base64:encode(<<"foo">>)},
+         {<<"state_hash">>, base64:encode(<<"foo">>)},
+         {<<"txs_hash">>, base64:encode(<<"foo">>)},
+         {<<"txs_hash">>, base64:encode(<<"foo">>)},
+         {<<"target">>, 42},
+         {<<"nonce">>, 42},
+         {<<"time">>, 42},
+         {<<"version">>, 42},
+         {<<"pow">>, lists:seq(1, 42)},
+         {<<"transactions">>, []}
+        ]),
+
+    lists:foreach(
+        fun({BrokenField, BlockMap}) ->
+            ct:log("Testing with a broken ~p", [BrokenField]),
+            {ok, Block} = aec_blocks:deserialize_from_map(BlockMap),
+            {ok, 200, _} = post_block(Block),
```

Comment:
The return code of a erronous `post_block` shouldn't be `200`, this will be addressed in [this PT](https://www.pivotaltracker.com/story/show/153629617) as well as other error codes.